src/Facets/UnitFacet.sol

```diff
@@ -0,0 +1,220 @@
+    /// @notice The address of the backend signer that is authorized to sign the UnitPayload
+    address internal immutable BACKEND_SIGNER;
+
+    /// Types ///
+
+    /// @notice The data that is signed by the backend
+    /// @param depositAddress The address to deposit the assets to
+    /// @param signature The signature of the UnitPayload signed by the backend signer using EIP-712 standard
+    /// @param deadline The deadline for the signature
+    struct UnitData {
+        address depositAddress;
+        bytes signature;
+        uint256 deadline;
+    }
+
+    /// Errors ///
+    /// @notice Thrown when the signature is invalid
+    error InvalidSignature();
+    /// @notice Thrown when the signature is expired
+    error SignatureExpired();
+
+    /// Constructor ///
+    /// @notice Initializes the UnitFacet contract
+    /// @param _backendSigner The address of the backend signer
+    constructor(address _backendSigner) {
+        if (_backendSigner == address(0)) {
+            revert InvalidConfig();
+        }
+        BACKEND_SIGNER = _backendSigner;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Unit
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _unitData Data specific to Unit
+    function startBridgeTokensViaUnit(
+        ILiFi.BridgeData memory _bridgeData,
+        UnitData calldata _unitData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        onlyAllowSourceToken(_bridgeData, LibAsset.NULL_ADDRESS) // only allow native asset
+        onlyAllowDestinationChain(_bridgeData, LIFI_CHAIN_ID_HYPERCORE)
+    {
+        _verifySignature(_bridgeData, _unitData);
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _unitData);
+    }
+
+    /// @notice Performs a swap before bridging via Unit
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _unitData Data specific to Unit
+    function swapAndStartBridgeTokensViaUnit(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        UnitData calldata _unitData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+        onlyAllowSourceToken(_bridgeData, LibAsset.NULL_ADDRESS) // only allow native asset
+        onlyAllowDestinationChain(_bridgeData, LIFI_CHAIN_ID_HYPERCORE)
+    {
+        _validateSwapOutputIsNative(_swapData);
```

Comment:
You are running `onlyAllowSourceToken` (which checks `bridgeData.sendingAssetId`) and `__validateSwapOutputIsNative` which checks the last output token.
I think it would be enough to run `onlyAllowSourceToken`. If the swap by accident yields the wrong (non-native) token, then the tx should fail anyway due to insufficient native balance

Reply:
that's true, I will remove `__validateSwapOutputIsNative` then